applications/FluidDynamicsApplication/python_scripts/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_symbolic_generator.py

```diff
@@ -57,19 +59,45 @@ def __init__(self, settings):
 
         self.template_filename = settings["template_filename"].GetString()
         self.output_filename = settings["output_filename"].GetString()
+
+        self.outstring = None
         self._GenerateFiles()
 
         self._print(2, settings)
 
+    def _FindAndRemoveIndentation(self, target_substring):
+        end = self.outstring.find(target_substring)
+        begin = end
+        leading_spaces = 0
+        while begin > 0:
+            if self.outstring[begin] == '\n':
+                break
+            if self.outstring[begin] != ' ':
+                end = begin - 1
+            begin -= 1
+
+        leading_spaces = end - begin
+
+        if leading_spaces % 4 == 0:
+            self.outstring = self.outstring[:begin+1] + self.outstring[end+1:]
+            return int(leading_spaces / 4)
+
+        raise ValueError(
+            "Inconsistent indentation in source file! Substitution {} found {} leading spaces, which is not a multiple of four.".format(
+            target_substring, leading_spaces
+        ))
+
     def _CollectAndReplace(self, target_substring, expression, name):
-        out = KratosSympy.OutputVector_CollectingFactors(expression, name, self.write_mode, replace_indices=False, assignment_op=" = ")
+        indentation_level  = self._FindAndRemoveIndentation(target_substring)
+
+        out = KratosSympy.OutputVector_CollectingFactors(expression, name, self.write_language, indentation_level=indentation_level, replace_indices=False, assignment_op=" = ")
         self.outstring = self.outstring.replace(target_substring, out)
```

Comment:
These might be useful for other elements. Maybe we can move them to the `sympy_fe_utilities.py` in the future.

Reply:
Agreed.

I left them here for now since it is quite inefficient (a very long string is copied in line 82, only to be copied again in line 94), so it should be used sparingly. If we want to put it in the core we might want to be smarter about it.